### PR TITLE
Update posts endpoint and testing

### PR DESCRIPTION
**Pull request by mkrumholz**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Linter has passed locally and any fixes were made for failures
- [ ] Travis build is passing
- [x] All tests passing locally

## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: Posts endpoint was not tested and incomplete

## What is the new behavior?

- Completes basic posts endpoint with testing for the posts query
- Adds testing for Post and User Types
- Updates factories to generate test data with images of variable sizes and appropriate subject matter
- Enables rack-cors gem for some added security

## Other information

This PR is JUST for the end point to get ALL posts, it does not yet include the option to get posts for a single user -- I will add that tomorrow.

### CODE REVIEW

**Review by JoePeterson51 — APPROVED** [review]

File: config/initializers/cors.rb [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
-#     origins 'example.com'
-#
-#     resource '*',
-#       headers: :any,
-#       methods: [:get, :post, :put, :patch, :delete, :options, :head]
-#   end
-# end
+Rails.application.config.middleware.insert_before 0, Rack::Cors do
+  allow do
+    origins 'localhost:3000', 'hivemindapp.netlify.app'
```
Comment (clairefields15) [inline config/initializers/cors.rb:10]: Yay!!

File: spec/factories/posts.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@
   factory :post do
     title { Faker::Books::Lovecraft.sentence }
     description { Faker::Books::Lovecraft.paragraph }
-    image { Faker::LoremFlickr.image }
+    image { Faker::LoremFlickr.image(size: "#{rand(200..800)}x#{rand(200..800)}", search_terms: ['bees', 'beehive']) }
```
Comment (clairefields15) [inline spec/factories/posts.rb:5]: Nice, thanks for adding the random sizer!

File: spec/factories/users.rb [hunk 3]
```diff
@@ -3,6 +3,6 @@
     username { Faker::Fantasy::Tolkien.character }
     region { Faker::Fantasy::Tolkien.location }
     biography { Faker::Fantasy::Tolkien.poem }
-    image { Faker::LoremFlickr.image }
+    avatar { Faker::LoremFlickr.image(size: "#{rand(100..250)}x#{rand(100..250)}", search_terms: ['bear']) }
```
Comment (clairefields15) [inline spec/factories/users.rb:6]: 😂 all bears, I love it

File: spec/graphql/types/query_type/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      expect(data.first[:createdAt]).to be_a String
+      expect(data.first[:user]).to eq({ id: user1.id.to_s, username: user1.username, avatar: user1.avatar })
+    end
+
+    it 'returns post type objects' do
+      expect(subject).to have_field(:posts).of_type('[Post!]!')
+    end
+
+    def query
+      <<~GQL
```
Comment (clairefields15) [inline spec/graphql/types/query_type/posts_spec.rb:38]: Nice! Can't wait to use this query!

**Review by clairefields15 — APPROVED** [review]
Looks great, excited to use these queries!

**Pull request merged**